support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsRequestHeaderAction.java

```diff
@@ -54,4 +59,10 @@ protected Credential constructCredentialsFromRequest(final RequestContext contex
         }
         return null;
     }
+
+    @Override
+    protected void onError(final RequestContext requestContext) {
+        WebUtils.putCasLoginFormViewable(requestContext, casProperties.getAuthn().getX509().isMixedMode());
```

Comment:
You might need to get the existing value and OR it with the setting here.

Reply:
Should that be done for the both X509 credential actions?

Reply:
I'd say so, yes. It's an oversight if they are not doing that already.

Reply:
I added a new method to WebUtils b/c `isCasLoginFormViewable` was defaulting to TRUE, which is good for use in UI, but not in OR statement.